crates/ark/src/data_explorer/r_data_explorer.rs

```diff
@@ -859,6 +859,110 @@ impl RDataExplorer {
         self.view_indices = Some(view_indices);
     }
 
+    /// Search the schema for columns matching the given filters and sort order.
+    ///
+    /// - `params`: The search parameters including filters and sort order.
+    fn search_schema(
+        &self,
+        params: amalthea::comm::data_explorer_comm::SearchSchemaParams,
+    ) -> anyhow::Result<DataExplorerBackendReply> {
+        let mut matching_indices: Vec<i64> = Vec::new();
+
+        // Get all column schemas
+        let all_columns = &self.shape.columns;
+
+        // Apply column filters to find matching columns
+        for (index, column) in all_columns.iter().enumerate() {
+            let column_index = index as i64;
+            let mut matches = true;
+
+            // Apply each filter to the column
+            for filter in &params.filters {
+                if !self.column_matches_filter(column, filter) {
+                    matches = false;
+                    break;
+                }
+            }
+
+            if matches {
+                matching_indices.push(column_index);
+            }
+        }
+
+        // Apply sort order
+        match params.sort_order {
+            amalthea::comm::data_explorer_comm::SearchSchemaSortOrder::Original => {
+                // matching_indices is already in original order
+            },
+            amalthea::comm::data_explorer_comm::SearchSchemaSortOrder::Ascending => {
+                matching_indices.sort_by(|&a, &b| {
+                    all_columns[a as usize]
+                        .column_name
+                        .cmp(&all_columns[b as usize].column_name)
+                });
+            },
+            amalthea::comm::data_explorer_comm::SearchSchemaSortOrder::Descending => {
+                matching_indices.sort_by(|&a, &b| {
+                    all_columns[b as usize]
+                        .column_name
+                        .cmp(&all_columns[a as usize].column_name)
```

Comment:
Slightly clearer? Since this is now the same as the previous branch except for the reverse()

```suggestion
                matching_indices.sort_by(|&a, &b| {
                    all_columns[a as usize]
                        .column_name
                        .cmp(&all_columns[b as usize].column_name)
                        .reverse()

```

Oh or maybe something like this:

```rust
match params.sort_order {
    amalthea::comm::data_explorer_comm::SearchSchemaSortOrder::Original => {
        // matching_indices is already in original order
    },
    order => {
        let ascending = matches!(
            order,
            amalthea::comm::data_explorer_comm::SearchSchemaSortOrder::Ascending
        );
        matching_indices.sort_by(|&a, &b| {
            let ord = all_columns[a as usize]
                .column_name
                .cmp(&all_columns[b as usize].column_name);
            if ascending { ord } else { ord.reverse() }
        });
    }
}
```